src/main/java/com/osp/dragon/service/ArticleService.java

```diff
@@ -15,16 +15,15 @@
 
 @Service
 public class ArticleService {
-
-
-    List<Article> articlesList;
+    private List<Article> articlesList;
 
     public ArticleService() throws IOException, ClassNotFoundException {
-        this.articlesList = initializeArticles();
+        ArticlesReader articlesReader = new ArticlesReader();
```

Comment:
To nic nie naprawia, konstruktor jest wywoływany raz. Więc tylko raz wczytasz artykuły do pamięci, a co dalej? Jak ktoś doda artykuł to jak go wczytasz? Jak tylko raz je wczytujesz? Artykuły powinny być wczytywane za każdym razem jak ktoś zawoła: załaduj artykuły